plugin/src/Reconciler/setProperty.lua

```diff
@@ -0,0 +1,48 @@
+--[[
+	Attempts to set a property on the given instance.
+]]
+
+local RbxDom = require(script.Parent.Parent.Parent.RbxDom)
+local Log = require(script.Parent.Parent.Parent.Log)
+local Error = require(script.Parent.Error)
+
+local function setProperty(instance, propertyName, value)
+	local descriptor = RbxDom.findCanonicalPropertyDescriptor(instance.ClassName, propertyName)
+
+	-- We can skip unknown properties; they're not likely reflected to Lua.
+	--
+	-- A good example of a property like this is `Model.ModelInPrimary`, which
+	-- is serialized but not reflected to Lua.
+	if descriptor == nil then
+		Log.trace("Skipping unknown property {}.{}", instance.ClassName, propertyName)
+
+		return true
+	end
+
+	if descriptor.scriptability == "None" or descriptor.scriptability == "Read" then
+		return false, Error.new(Error.UnwritableProperty, {
+			className = instance.ClassName,
+			propertyName = propertyName,
+		})
+	end
+
+	local ok, err = descriptor:write(instance, value)
+
+	if not ok then
+		if err.kind == RbxDom.Error.Kind.Roblox and err.extra:find("lacking permission") then
+			return false, Error.new(Error.LackingPropertyPermissions, {
+				className = instance.ClassName,
+				propertyName = propertyName,
+			})
+		end
+
+		return false, Error.new(Error.OtherPropertyError, {
+			className = instance.ClassName,
+			propertyName = propertyName,
+		})
+	end
+
+	return true
+end
+
+return setProperty
```

Comment:
`getProperty` and `setProperty` both share a lot of functionality, is there a way we could have more code reuse between them? I'm not sure if it's the best way but I tackled a similar problem by making a function like `propertyOperation("Write", instance, propertyName, value)`. What do you think about this kind of approach?

Reply:
I think the similarities are pretty much just the error handling boilerplate. Maybe cutting down on that would make these functions feel less similar?